src/components/Screen.tsx

```diff
@@ -271,8 +272,12 @@ export const InnerScreen = React.forwardRef<View, ScreenProps>(
         }
       };
 
+      const freeze =
+        freezeOnBlur &&
+        (isNativeStack ? Boolean(shouldFreeze) : activityState === 0);
```

Comment:
Maybe should always use `shouldFreeze` if a boolean is passed? I think for JS stack `activityState` will be `0` for preloaded screens as well so we will need a way for preloaded screens to opt-out.

Reply:
Ok, looking at it: https://github.com/react-navigation/react-navigation/blob/b70823fa58ea36217534e646f172516665b684a0/packages/stack/src/views/Stack/CardStack.tsx#L699-L701 it makes sense. But then we should add the `shouldFreeze` logic to stack too and remove relying on `activityState` at all.

Reply:
@satya164 removing the logic would be breaking when using screens with react navigation 6 (same for native stack), so it's upto you

Reply:
So what is the official version of support? I heard that RNScreens v4 should be working only with navigation v7.

Reply:
And v6 is expected to not work correctly with v4

Reply:
There's no breaking change in screens v4 that should affect React Navigation 6 JS Stack. There are some changes in header for native stack intended to fix some bugs but from my testing it doesn't break anything either - though I guess the bugs it's supposed to fix will still be there, but it's just keeping same behavior.